packages/flutter_tools/lib/src/runner/flutter_command_runner.dart

```diff
@@ -147,7 +147,7 @@ class FlutterCommandRunner extends CommandRunner {
     // enginePath's initialiser uses it).
     Cache.flutterRoot = path.normalize(path.absolute(globalResults['flutter-root']));
 
-    if (globalResults['lock']);
+    if (globalResults['lock'] && !(Platform.environment['FLUTTER_ALREADY_LOCKED'] == 'true'))
```

Comment:
If you pass `allowReentrantFlutter: true` in https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/commands/upgrade.dart#L54, you don't need the lock flag anymore, right?